isis/tests/FunctionalTestsJigsaw.cpp

```diff
@@ -0,0 +1,104 @@
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "ControlNet.h"
+#include "Statistics.h"
+
+#include "jigsaw.h"
+
+#include "gtest/gtest.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/jigsaw.xml").expanded();
+
+TEST_F(ObservationPair, FunctionalTestJigsawCamSolveAll) {
+  // delete to remove old camera for when cam is updated
+  delete cubeL;
+  delete cubeR;      
+  
+  QTemporaryDir prefix;
+  QString outCnetFileName = prefix.path() + "/outTemp.net";
+  QVector<QString> args = {"fromlist="+cubeListFile, "cnet="+cnetPath, "onet="+outCnetFileName, 
+                           "observations=yes", "update=yes", "Cksolvedegree=3", 
+                           "Camsolve=all", "twist=no", "Spsolve=none", "Radius=no", "imagescsv=on", "file_prefix="+prefix.path()};
+
+  UserInterface options(APP_XML, args);
+  
+  Pvl log; 
+  try {
+    jigsaw(options, &log);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to bundle: " << e.what() << std::endl;
+  }
+  
+  // images were updated 
+  cubeL = new Cube(cubeLPath, "r");
+  cubeR = new Cube(cubeRPath, "r");
```

Comment:
You could close the Cube files and then reopen them here instead of having to delete and re-create.

Reply:
for some reason they don't get updated correctly if you don't delete them before passing them in (seems to a thing with these apps that pass in images through a filelist?).